[PAB-4624] - Improve documentation around apama-ctrl variants (#2447)
SoftwareAG/c8y-docs: content/streaming-analytics/introduction-analytics-bundle/microservice-and-applications.md

title: Microservice runtime and applications
layout: redirect
---

> Analytic models, EPL apps and smart rules are executed in an Apama-ctrl microservice. All microservices come with pre-configured resource limits (example: CPU and Memory). Resource usage of analytics models, EPL apps and smart rules varies on various factors like application complexity, number of devices etc. It's advisable for the customer to profile their application to prevent out of memory issues.

Reviewer: ```suggestion
Analytic models, EPL apps and smart rules are executed in an Apama-ctrl microservice. All microservices come with pre-configured resource limits (for example, CPU and memory). The resource usage of analytic models, EPL apps and smart rules depends on various factors such as application complexity, number of devices, etc. It is recommended that customers profile their applications to avoid out-of-memory issues.
```

| To do this                  | you need the following                                      |
| --------------------------- | ----------------------------------------------------------- |

> | Build analytic models       | Apama-ctrl microservice variant (example: apama-ctrl-starter) and Streaming Analytics application |

Reviewer: ```suggestion
| Build analytic models       | Apama-ctrl microservice variant (for example, Apama-ctrl-starter) and Streaming Analytics application |
```

> | Develop EPL apps            | Apama-ctrl microservice variant (example: apama-ctrl-250mc-1g) and Streaming Analytics application |

Reviewer: ```suggestion
| Develop EPL apps            | Apama-ctrl microservice variant (for example, Apama-ctrl-250mc-1g) and Streaming Analytics application |
```

| Use smart rules             | Any Apama-ctrl microservice variant and Smartrule microservice (included in {{< product-c8y-iot >}}'s {{< standard-tenant >}}) |

See also the following sections:
* [Subscribed applications](/standard-tenant/ecosystem/#subscribed-applications)
* [Subscribed microservices](/standard-tenant/ecosystem/#subscribed-microservices)
* [Smart rules](/cockpit/smart-rules/)

If your tenant is subscribed to the Apama-ctrl-starter microservice, then the following applies:
- Limited number of at most 3 active analytic models. Custom blocks written with the Analytics Builder Block SDK cannot be used.
- The **EPL Apps** page is not available in the Streaming Analytics application.

> - Supports Smart rules.

Reviewer: ```suggestion
- Smart rules are supported.
```

If your tenant is subscribed to the Apama-ctrl-250mc-1g  microservice (or one of its larger variants), then the following applies:

> - Supports Smart rules, **Analytics Builder** and **EPL Apps**

Reviewer: ```suggestion
- Unlimited number of analytic models. Custom blocks written with the Analytics Builder Block SDK can be used.
- EPL apps are enabled.
- Smart rules are supported.
```

Reviewer: The wording should be similar in all cases. That's why I suggest to split this up into several bullets.

Reviewer: However, if you do not want to commit this suggestion, then you should go with this (without any bold markup):

Smart rules, analytic models and EPL apps are supported.

Author: See 7216ff615, does this work?

Reviewer: Yes, that's good :-)
But I forgot: I normally put smart rules at the end of an enumeration because they are maintained in a different place, not within the Streaming Analytics application. Therefore, I'd move that to the end:

Analytic models, EPL apps and smart rules are supported.

Author: done 7b99ff1c9

If your tenant is subscribed to the Apama-ctrl-mt-4c-16g microservice, then the following applies:
- Multi-tenant support.
- EPL apps are only enabled on the tenant that owns the microservice, but disabled on the subtenants.
- The **Analytics Builder** page is currently not available in the Streaming Analytics application.

> - Supports Smart rules.

Reviewer: Smart rules are supported.

> - Supports Smart rules.

Reviewer: ```suggestion
- Smart rules are supported.
```

> If your tenant is subscribed to the Apama-ctrl-smartrules or Apama-ctrl-smartrulesmt microservice

Reviewer: ```suggestion
If your tenant is subscribed to the Apama-ctrl-smartrules or Apama-ctrl-smartrulesmt microservice, then the following applies:
```

> - Supports Smart rules.

Reviewer: ```suggestion
- Smart rules are supported.
```

| To do this                  | you need the following                                      |
| --------------------------- | ----------------------------------------------------------- |
| Build analytic models       | Apama-ctrl microservice variant (for example, Apama-ctrl-starter) and Streaming Analytics application |
| Develop EPL apps            | Apama-ctrl microservice variant (for example, Apama-ctrl-250mc-1g) and Streaming Analytics application |
| Use smart rules             | Any Apama-ctrl microservice variant and Smartrule microservice (included in {{< product-c8y-iot >}}'s {{< standard-tenant >}}) |

See also the following sections:
* [Subscribed applications](/standard-tenant/ecosystem/#subscribed-applications)
* [Subscribed microservices](/standard-tenant/ecosystem/#subscribed-microservices)
* [Smart rules](/cockpit/smart-rules/)

If your tenant is subscribed to the Apama-ctrl-starter microservice, then the following applies:
- Limited number of at most 3 active analytic models. Custom blocks written with the Analytics Builder Block SDK cannot be used.
- The **EPL Apps** page is not available in the Streaming Analytics application.
- Smart rules are supported.

> If your tenant is subscribed to the Apama-ctrl-250mc-1g  microservice (or one of its larger variants), then the following applies:

Reviewer: ```suggestion
If your tenant is subscribed to the Apama-ctrl-250mc-1g microservice (or one of its larger variants), then the following applies:
```

Reviewer: Only saw this superfluous blank now, sorry. Otherwise all good now :-)
